src/main/java/org/kiwiproject/registry/consul/server/ConsulRegistryService.java

```diff
@@ -0,0 +1,220 @@
+
+        var registration = fromServiceInstance(serviceToRegister);
+
+        var service = registerRetryer.tryGetObject(() -> {
+            consul.agentClient().register(registration);
+            return serviceToRegister.withInstanceId(registration.getId());
+        });
+
+        if (service.isEmpty()) {
+            LOG.error("Registration failed for service {} with id {}. See logs above", serviceToRegister.getServiceName(), registration.getId());
+            var errMsg = format("Unable to register service %s, id %s with Consul after %s attempts",
+                    serviceToRegister.getServiceName(), registration.getId(), MAX_REGISTRATION_ATTEMPTS);
+            throw new RegistrationException(errMsg);
+        }
+
+        registeredService.set(service.get());
+        return registeredService.get().toBuilder().build();
+    }
+
+    @Override
+    public ServiceInstance updateStatus(ServiceInstance.Status newStatus) {
+        LOG.info("Ignoring status update as Consul handles it's own status through it's health check process");
+
+        if (isRegistered()) {
+            return registeredService.get().toBuilder().build();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void unregister() {
+        checkState(isRegistered(), "Cannot unregister since registration was never called");
+
+        var serviceName = registeredService.get().getServiceName();
+        var instanceId = registeredService.get().getInstanceId();
+
+        LOG.info("Unregistering service {} with id {}", serviceName, instanceId);
+
+        var result = unregisterRetryer.tryGetObject(() -> {
+            consul.agentClient().deregister(registeredService.get().getInstanceId());
+            return instanceId;
+        });
+
+        Optionals.ifPresentOrElseThrow(result,
+                id -> {
+                    LOG.info("Service with name {} and id {} has been unregistered successfully", serviceName, instanceId);
+                    registeredService.set(null);
+                },
+                () -> {
+                    var msg = format("Error un-registering service {}, id {}", serviceName, instanceId);
+                    LOG.error(msg);
+                    return new RegistrationException(msg);
+                });
+    }
+
+    private boolean isRegistered() {
+        return nonNull(registeredService.get());
+    }
+
+    private Registration fromServiceInstance(ServiceInstance serviceInstance) {
+
+        var check = ImmutableRegCheck.builder()
+                .http(urlForPath(serviceInstance.getHostName(), serviceInstance.getPorts(), Port.PortType.ADMIN,
+                        serviceInstance.getPaths().getStatusPath()))
+                .interval(String.format("%ds", config.getCheckIntervalInSeconds()))
+                .deregisterCriticalServiceAfter(
+                        String.format("%dm", config.getDeregisterIntervalInMinutes()))
```

Comment:
Suggest using `KiwiStrings.format` (or `f`) here instead of `String.format`. You can then just use the `{}` (or `%s`)

Reply:
Done